File: src/main/java/net/atos/zac/documentcreation/model/DocumentCreationData.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.documentcreation.model;

import net.atos.client.zgw.drc.model.generated.StatusEnum;
import net.atos.client.zgw.zrc.model.Zaak;
import net.atos.client.zgw.ztc.model.generated.InformatieObjectType;

public class DocumentCreationData {
    private final Zaak zaak;
    private final String taskId;
    private final StatusEnum informatieobjectStatus = StatusEnum.TER_VASTSTELLING;
    private final InformatieObjectType informatieobjecttype;

    public DocumentCreationData(
            final Zaak zaak,
            final String taskId,
            final InformatieObjectType informatieobjecttype
    ) {
        this.zaak = zaak;
        this.taskId = taskId;
        this.informatieobjecttype = informatieobjecttype;
    }

    public Zaak getZaak() {
        return zaak;
    }

    public StatusEnum getInformatieobjectStatus() {
        return informatieobjectStatus;
    }

    public InformatieObjectType getInformatieobjecttype() {
        return informatieobjecttype;
    }

    public String getTaskId() {
        return taskId;
    }
}

File: src/main/kotlin/net/atos/zac/documentcreation/model/DocumentCreationData.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.documentcreation.model

import net.atos.client.zgw.zrc.model.Zaak
import net.atos.client.zgw.ztc.model.generated.InformatieObjectType

data class DocumentCreationData(
    val zaak: Zaak,

    val taskId: String? = null,

    @Deprecated(
        "Only required for the SmartDocuments unattended flow. " +
            "Will be removed in future once we no longer support the SmartDocuments attended flow."
    )
    val informatieobjecttype: InformatieObjectType? = null,

    /**
     * Nullable for now but once we no longer support the SmartDocuments attended flow we should make this required
     */
    val templateGroup: String? = null,

    /**
     * Nullable for now but once we no longer support the SmartDocuments attended flow we should make this required
     */
    val template: String? = null
)
